src/common.rs

```diff
@@ -524,6 +524,46 @@ bitflags! {
     }
 }
 
+#[derive(Debug)]
+pub struct ParameterError {
+    inner: ParameterErrorKind,
+}
+
+#[derive(Debug)]
+pub(crate) enum ParameterErrorKind {
+    /// A provided buffer must be large enough to hold the image data. Where the buffer can be
```

Comment:
This error can be triggered by a buffer that is either too large or too small, so we should probably start the comment with a sentence indicating that.